Tools/Scripts/libraries/webkitcorepy/webkitcorepy/credentials.py

```diff
@@ -83,13 +83,40 @@ def credentials(url, required=True, name=None, prompt=None, key_name='password')
         _cache[name] = (username, key)
 
     if keyring and (username_prompted or key_prompted):
-        sys.stderr.write('Store username and {} in system keyring for {}? (Y/N): '.format(key_name, url))
-        response = Terminal.input()
-        if response.lower() in ['y', 'yes', 'ok']:
+        if Terminal.choose(
+            'Store username and {} in system keyring for {}?'.format(key_name, url),
+            default='Yes',
+        ) == 'Yes':
             sys.stderr.write('Storing credentials...\n')
             keyring.set_password(url, 'username', username)
             keyring.set_password(url, username, key)
         else:
             sys.stderr.write('Credentials cached in process.\n')
 
     return username, key
+
+
+def delete_credentials(url, name=None):
+    global _cache
+
+    name = name or url.split('/')[2].replace('.', '_')
+    if name in _cache:
+        del _cache[name]
+
+    with OutputCapture():
+        try:
+            import keyring
+
+            username = None
+            try:
+                if keyring:
+                    username = keyring.get_password(url, 'username')
```

Comment:
is it storing username or password?

Reply:
This function is all deleting, and this first part is explicitly username.

Some background on this: for bugzilla and SVN, the user's email address was the username. Which made finding credentials really easy because we could query the global git settings to find the username. This doesn't work so well for GitHub because emails and usernames don't necessarily have a relationship. The solution I came up with for this was to store the username for a given account in the keychain keyed by `username` along with password keyed by the actual username

Reply:
ok